lib/Sema/TypeCheckDecl.cpp

```diff
@@ -4306,9 +4306,6 @@ void TypeChecker::validateDecl(ValueDecl *D) {
 
     checkDeclAttributesEarly(DD);
 
-    if (auto enclosingClass = dyn_cast<ClassDecl>(DD->getDeclContext()))
-      DD->copyFormalAccessFrom(enclosingClass, /*sourceIsParentContext*/true);
```

Comment:
Propagating `@usableFromInline` has to change from a 'push' model (done in validateDecl()) to a 'pull' model (done when you first query access on the destructor). Until then, note that an earlier patch in the PR changes ValueDecl::isUsableFromInline() to ask the class for the attribute if called on a destructor. This is OK because destructors themselves don't allow the attribute to be written in source -- its always taken from the class.

Reply:
It's a later patch in the series, not an earlier one, but I see what you mean about where this is going. If destructors were always synthesized, it would make sense to stick to the push model, but they're not. Thanks for the explanation!

Reply:
GitHub shows them in the wrong order unfortunately, because it seems to sort by date and not by the actual order in the repo. I think that's a bug...